### PR TITLE
[NN] Add EdgeGAT operator

### PR DESCRIPTION
**Pull request by schmidt-ju**
## Description
Hey,

I added the EdgeGAT operator of the Robotics and Automation Letters (RA-L) paper [SCENE: Reasoning about Traffic Scenes using Heterogeneous Graph Neural Networks](https://arxiv.org/abs/2301.03512).
The already available [EGAT](https://docs.dgl.ai/en/0.9.x/generated/dgl.nn.pytorch.conv.EGATConv.html) operator does only include edge feature for the calculation of the attention coefficients.

The operator of this PR, namely EdgeGAT, is more advanced: It also includes the edge features directly for the node update and not only for the calculation of the attention coefficients.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

Looking forward to hearing from you!

Julian

### CODE REVIEW

**Comment by schmidt-ju** [comment]
Hey,

I am not sure, why the CI is failing. I just recreated the test cases and they seem to work on my local machine.

Also, I am not able to access the produced logs.
There is probably something that I am missing right now. Can you help me?

Julian

**Comment by BarclayII** [comment]
@dgl-bot

**Comment by BarclayII** [comment]
@schmidt-ju Hi Julian, currently CI requires authentication from one of the core developers so someone from us has to invoke the CI manually.  And after the CI is invoked and an error other than "Authentication failed" is shown, you will have to click on the `Report path` link and find the logs like below:
image
Looks like there is lint issues.  Could you take a look?

Please feel free to nudge us if you have any problems.  Thanks!

**Comment by schmidt-ju** [comment]
Hey,

thanks for the information!
I fixed the lint issues (to the best of my knowledge^^). At least on my local machine the `task_lint.sh` script does not throw errors anymore.

**Comment by BarclayII** [comment]
Seems that there's still lint issues.  You could try [`lintrunner -a`](https://pypi.org/project/lintrunner/) to apply the lint fixes automatically.

**Comment by BarclayII** [comment]
The files are `python/dgl/nn/pytorch/conv/edgegatconv.py` and `python/dgl/nn/pytorch/conv/__init__.py`.

**Comment by schmidt-ju** [comment]
Hey @BarclayII,

I fixed the lint issues in the two mentioned files using lintrunner.
According to the "Checks" tab, there are still some issues in `tests/pytorch/test_nn.py`.

My modifications in this file are limited to two new test methods. The linter, however, shows errors in the whole file and not limited to my small changes.
Is this intentional?
I can reformat the whole file, but I don't think that this is something you want.

Looking forward to hearing from you.

Julian

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by frozenbugs** [comment]
@schmidt-ju  Can you fix the conflict, I think the lint issue should be addressed.

**Comment by schmidt-ju** [comment]
Hey @frozenbugs,
I resolved the conflicts. Lint issues are now gone, too (because of the changes in the test_nn.py file).

**Comment by frozenbugs** [comment]
@dgl-bot

File: python/dgl/nn/pytorch/conv/edgegatconv.py
Comment (frozenbugs) [inline python/dgl/nn/pytorch/conv/edgegatconv.py:317]: super nit: capitalize the L and end with . for all comments.

File: python/dgl/nn/pytorch/conv/edgegatconv.py
Comment (frozenbugs) [inline python/dgl/nn/pytorch/conv/edgegatconv.py:339]: Avoiding usage of "we", "you"? Make it a statement.

File: python/dgl/nn/pytorch/conv/edgegatconv.py
Comment (frozenbugs) [inline python/dgl/nn/pytorch/conv/edgegatconv.py:12]: Maybe mention SCENE in the name of EdgeGATConv? Since we already have another similar egatconv https://github.com/dmlc/dgl/blob/master/python/dgl/nn/pytorch/conv/egatconv.py

File: python/dgl/nn/pytorch/conv/edgegatconv.py
Comment (BarclayII) [inline python/dgl/nn/pytorch/conv/edgegatconv.py:16]: A blank line is needed under `.. math::` block.

File: python/dgl/nn/pytorch/conv/edgegatconv.py
Comment (BarclayII) [inline python/dgl/nn/pytorch/conv/edgegatconv.py:25]: Ditto

**Review by BarclayII — APPROVED** [review]
LGTM, we can merge it after the code style comments are addressed.

**Comment by frozenbugs** [comment]
Overall LGTM, thanks for using and contributing to DGL. @BarclayII will take another look then approve.

**Comment by schmidt-ju** [comment]
Hey @BarclayII,

thanks for your overall positive feedback.

I addressed all of your comments.
The only thing left is @frozenbugs comment:
> Maybe mention SCENE in the name of EdgeGATConv? Since we already have another similar egatconv https://github.com/dmlc/dgl/blob/master/python/dgl/nn/pytorch/conv/egatconv.py

I have talked to my co-author about this and we would like to leave the name as it is.
This makes the name in the paper and in DGL consistent with each other.

Looking forward to hearing from you!


Best

Julian

**Review by frozenbugs — APPROVED** [review]

**Comment by BarclayII** [comment]
@dgl-bot

**Comment by schmidt-ju** [comment]
Hey @BarclayII,

do you already have a rough estimate of when the merge into the main branch will be carried out?

Julian

**Comment by frozenbugs** [comment]
> Hey @BarclayII,
> 
> do you already have a rough estimate of when the merge into the main branch will be carried out?
> 
> Julian

Hi Julian, sorry for the trouble, just merged the PR.
(I am new to github, and I thought you can merge also long as we approve)

**Pull request merged**
